### PR TITLE
Implements what described in #6688

### PR DESCRIPTION
**Pull request by HermesSbicego-Laser**
Supported Syntaxes for Request and Form tokens are:
1. QueryString:(param1) or Form:(param1)
2. QueryString:param1 or Form:param1
3. QueryString:(param1).SomeOtherTextToken or Form:(param1).SomeOtherTextToken

If you want to Chain TextTokens you have to use the 3rd syntax
the element (here param1) has been surrounded with brackets in order to preserve backward compatibility.

### CODE REVIEW

File: src/Orchard.Web/Modules/Orchard.Tokens/Providers/RequestTokens.cs [hunk 1]
```diff
@@ -1,25 +1,30 @@
...
         }
 
         public Localizer T { get; set; }
 
         public void Describe(DescribeContext context) {
             context.For("Request", T("Http Request"), T("Current Http Request tokens."))
-                .Token("QueryString:*", T("QueryString:<element>"), T("The Query String value for the specified element."))
-                .Token("Form:*", T("Form:<element>"), T("The Form value for the specified element."))
+                .Token("QueryString:*", T("QueryString:<element>"), T("The Query String value for the specified element. If you want to chain text, surround the <element> with brackets [e.g. QueryString:(param1)]."))
+                .Token("Form:*", T("Form:<element>"), T("The Form value for the specified element. If you want to chain text, surround the <element> with brackets [e.g. Form:(param1)]."))
```
Comment (DanielStolt) [inline src/Orchard.Web/Modules/Orchard.Tokens/Providers/RequestTokens.cs]: I think these explanation strings are a bit confusing, as they actually contain brackets but what I think you are referring to are the parentheses. A better hint text in my opinion would be to remove the brackets around the example and use the word paretheses, like so:

`The query string value for the specified element. To chain text, surround the <element> with parentheses, e.g. 'QueryString:(param1)'.`

**Comment by sebastienros** [comment]
Can you run the unit tests and tell us if they all pass? The CI is broken right now.

**Comment by HermesSbicego-Laser** [comment]
@sebastienros I branched it from master. but I cannot run test on VS 2015. If I Rebuild solution, I obtain these errors, it seems to be a Orchard.JobsQueue	problem...
What should I do in order to run Tests?

> Severity	Code	Description	Project	File	Line	Suppression State
> Error	CS0234	The type or namespace name 'ContentManagement' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	3	Active
> Error	CS0234	The type or namespace name 'DisplayManagement' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	4	Active
> Error	CS0234	The type or namespace name 'Environment' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	5	Active
> Error	CS0234	The type or namespace name 'Localization' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	6	Active
> Error	CS0234	The type or namespace name 'Mvc' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	9	Active
> Error	CS0234	The type or namespace name 'UI' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	10	Active
> Error	CS0234	The type or namespace name 'UI' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	11	Active
> Error	CS0234	The type or namespace name 'UI' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	12	Active
> Error	CS0246	The type or namespace name 'AdminAttribute' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	15	Active
> Error	CS0246	The type or namespace name 'Admin' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	15	Active
> Error	CS0246	The type or namespace name 'OrchardFeatureAttribute' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	16	Active
> Error	CS0246	The type or namespace name 'OrchardFeature' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	16	Active
> Error	CS0246	The type or namespace name 'IOrchardServices' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	19	Active
> Error	CS0246	The type or namespace name 'IShapeFactory' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	24	Active
> Error	CS0246	The type or namespace name 'IOrchardServices' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	25	Active
> Error	CS0246	The type or namespace name 'Localizer' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	35	Active
> Error	CS0246	The type or namespace name 'PagerParameters' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	46	Active
> Error	CS0246	The type or namespace name 'FormValueRequiredAttribute' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	61	Active
> Error	CS0246	The type or namespace name 'FormValueRequired' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	61	Active
> Error	CS0246	The type or namespace name 'FormValueRequiredAttribute' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	67	Active
> Error	CS0246	The type or namespace name 'FormValueRequired' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	67	Active
> Error	CS0246	The type or namespace name 'FormValueRequiredAttribute' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	75	Active
> Error	CS0246	The type or namespace name 'FormValueRequired' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	75	Active
> Error	CS0246	The type or namespace name 'FormValueRequiredAttribute' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	83	Active
> Error	CS0246	The type or namespace name 'FormValueRequired' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Controllers\AdminController.cs	83	Active
> Error	CS0234	The type or namespace name 'ContentManagement' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Drivers\JobsQueueSettingsPartDriver.cs	1	Active
> Error	CS0234	The type or namespace name 'ContentManagement' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Drivers\JobsQueueSettingsPartDriver.cs	2	Active
> Error	CS0234	The type or namespace name 'Localization' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Drivers\JobsQueueSettingsPartDriver.cs	3	Active
> Error	CS0234	The type or namespace name 'Messaging' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Drivers\JobsQueueSettingsPartDriver.cs	6	Active
> Error	CS0246	The type or namespace name 'ContentPartDriver<>' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Drivers\JobsQueueSettingsPartDriver.cs	9	Active
> Error	CS0246	The type or namespace name 'IOrchardServices' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Drivers\JobsQueueSettingsPartDriver.cs	11	Active
> Error	CS0246	The type or namespace name 'IOrchardServices' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Drivers\JobsQueueSettingsPartDriver.cs	13	Active
> Error	CS0246	The type or namespace name 'Localizer' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Drivers\JobsQueueSettingsPartDriver.cs	18	Active
> Error	CS0115	'JobsQueueSettingsPartDriver.Prefix': no suitable method found to override	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Drivers\JobsQueueSettingsPartDriver.cs	20	Active
> Error	CS0246	The type or namespace name 'DriverResult' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Drivers\JobsQueueSettingsPartDriver.cs	22	Active
> Error	CS0115	'JobsQueueSettingsPartDriver.Editor(JobsQueueSettingsPart, dynamic)': no suitable method found to override	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Drivers\JobsQueueSettingsPartDriver.cs	22	Active
> Error	CS0246	The type or namespace name 'DriverResult' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Drivers\JobsQueueSettingsPartDriver.cs	31	Active
> Error	CS0115	'JobsQueueSettingsPartDriver.Editor(JobsQueueSettingsPart, IUpdateModel, dynamic)': no suitable method found to override	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Drivers\JobsQueueSettingsPartDriver.cs	31	Active
> Error	CS0246	The type or namespace name 'IUpdateModel' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Drivers\JobsQueueSettingsPartDriver.cs	31	Active
> Error	CS0234	The type or namespace name 'ContentManagement' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Handlers\JobsQueueSettingsPartHandler.cs	1	Active
> Error	CS0246	The type or namespace name 'ContentHandler' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Handlers\JobsQueueSettingsPartHandler.cs	5	Active
> Error	CS0234	The type or namespace name 'Data' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Migrations.cs	2	Active
> Error	CS0246	The type or namespace name 'DataMigrationImpl' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Migrations.cs	5	Active
> Error	CS0234	The type or namespace name 'ContentManagement' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Models\JobsQueueSettingsPart.cs	1	Active
> Error	CS0246	The type or namespace name 'ContentPart' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Models\JobsQueueSettingsPart.cs	4	Active
> Error	CS0234	The type or namespace name 'Data' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Models\QueuedJobRecord.cs	2	Active
> Error	CS0246	The type or namespace name 'StringLengthMaxAttribute' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Models\QueuedJobRecord.cs	10	Active
> Error	CS0246	The type or namespace name 'StringLengthMax' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Models\QueuedJobRecord.cs	10	Active
> Error	CS0246	The type or namespace name 'IDependency' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\IJobsQueueManager.cs	5	Active
> Error	CS0246	The type or namespace name 'ISingletonDependency' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\IJobsQueueProcessor.cs	2	Active
> Error	CS0234	The type or namespace name 'Events' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\IJobsQueueService.cs	1	Active
> Error	CS0246	The type or namespace name 'IEventHandler' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\IJobsQueueService.cs	5	Active
> Error	CS0234	The type or namespace name 'Tasks' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueBackgroundTask.cs	1	Active
> Error	CS0246	The type or namespace name 'Component' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueBackgroundTask.cs	4	Active
> Error	CS0246	The type or namespace name 'IBackgroundTask' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueBackgroundTask.cs	4	Active
> Error	CS0234	The type or namespace name 'ContentManagement' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueManager.cs	4	Active
> Error	CS0234	The type or namespace name 'Data' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueManager.cs	5	Active
> Error	CS0234	The type or namespace name 'Settings' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueManager.cs	7	Active
> Error	CS0246	The type or namespace name 'IRepository<>' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueManager.cs	11	Active
> Error	CS0246	The type or namespace name 'IRepository<>' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueManager.cs	15	Active
> Error	CS0246	The type or namespace name 'ISiteService' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueManager.cs	16	Active
> Error	CS0234	The type or namespace name 'Environment' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueProcessor.cs	5	Active
> Error	CS0234	The type or namespace name 'Events' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueProcessor.cs	6	Active
> Error	CS0234	The type or namespace name 'Logging' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueProcessor.cs	8	Active
> Error	CS0234	The type or namespace name 'Tasks' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueProcessor.cs	9	Active
> Error	CS0246	The type or namespace name 'Work<>' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueProcessor.cs	13	Active
> Error	CS0246	The type or namespace name 'Work<>' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueProcessor.cs	14	Active
> Error	CS0246	The type or namespace name 'IEventBus' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueProcessor.cs	14	Active
> Error	CS0246	The type or namespace name 'Work<>' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueProcessor.cs	15	Active
> Error	CS0246	The type or namespace name 'IDistributedLockService' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueProcessor.cs	15	Active
> Error	CS0246	The type or namespace name 'Work<>' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueProcessor.cs	18	Active
> Error	CS0246	The type or namespace name 'Work<>' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueProcessor.cs	19	Active
> Error	CS0246	The type or namespace name 'IEventBus' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueProcessor.cs	19	Active
> Error	CS0246	The type or namespace name 'Work<>' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueProcessor.cs	20	Active
> Error	CS0246	The type or namespace name 'IDistributedLockService' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueProcessor.cs	20	Active
> Error	CS0246	The type or namespace name 'ILogger' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueProcessor.cs	28	Active
> Error	CS0234	The type or namespace name 'Data' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueService.cs	2	Active
> Error	CS0234	The type or namespace name 'Services' does not exist in the namespace 'Orchard' (are you missing an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueService.cs	4	Active
> Error	CS0246	The type or namespace name 'IClock' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueService.cs	8	Active
> Error	CS0246	The type or namespace name 'IRepository<>' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueService.cs	9	Active
> Error	CS0246	The type or namespace name 'IClock' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueService.cs	12	Active
> Error	CS0246	The type or namespace name 'IRepository<>' could not be found (are you missing a using directive or an assembly reference?)	Orchard.JobsQueue	C:\Sviluppo\Laser.Platform.Orchard\Orchard.Sources\src\Orchard.Web\Modules\Orchard.JobsQueue\Services\JobsQueueService.cs	13	Active
> 
>

**Comment by HermesSbicego-Laser** [comment]
Little update, if I merge 1.10.x into my branch I am able to Build but i still cannot run Tests... It seems VS 2015 does not recognize them as Tests, any idea of why?

**Comment by rtpHarry** [comment]
The specflow tests can be run by following these instructions:

 - https://github.com/OrchardCMS/OrchardDoc/issues/304

**Comment by HermesSbicego-Laser** [comment]
Thank you @rptHarry and @MatteoPiovanelli-Laser, after an Odissey I finally was able to run tests...
here the results from Tokens.Test
test results

File: src/Orchard.Web/Modules/Orchard.Tokens/Providers/RequestTokens.cs [hunk 2]
```diff
@@ -98,5 +113,53 @@ public class RequestTokens : ITokenProvider {
...
+            tokenPrefix = token.Substring(0, token.IndexOf(":"));
+            if (!_textChainableTokens.Contains(tokenPrefix, StringComparer.OrdinalIgnoreCase)) {
+                return null;
+            }
+
+            // use ")." as chars combination to discover the end of the parameter
+            chainIndex = token.IndexOf(").") + 1;
+            tokenLength = (tokenPrefix + ":").Length;
+            if (chainIndex == 0) {// ")." has not be found
+                return Regex.Replace(token.Substring(tokenLength), @"[\(|\)]", "");
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.Tokens/Providers/RequestTokens.cs]: Don't use a Regex to remove specific chars.

File: src/Orchard.Web/Modules/Orchard.Tokens/Providers/RequestTokens.cs [hunk 3]
```diff
@@ -98,5 +113,53 @@ public class RequestTokens : ITokenProvider {
...
+            tokenPrefix = token.Substring(0, token.IndexOf(":"));
+            if (!_textChainableTokens.Contains(tokenPrefix, StringComparer.OrdinalIgnoreCase)) {
+                return null;
+            }
+
+            // use ")." as chars combination to discover the end of the parameter
+            chainIndex = token.IndexOf(").") + 1;
+            tokenLength = (tokenPrefix + ":").Length;
+            if (chainIndex == 0) { // ")." has not be found
+                return new Tuple<string, string>(Regex.Replace(token.Substring(tokenLength), @"[\(|\)]", ""), token.Length.ToString());
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.Tokens/Providers/RequestTokens.cs]: Same. And also, if you reuse the same Regex then make one static and RegexOptions.Compiled.

Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.Tokens/Providers/RequestTokens.cs]: But not here, just remove it.

**Pull request merged**